nestkernel/node_manager.cpp

```diff
@@ -394,11 +391,19 @@ Node* NodeManager::get_node( index gid, thread t )
   return node;
 }
 
-Node* NodeManager::get_node_indp_thread( index gid )
+Node* NodeManager::get_node_or_proxy( index gid )
 {
   thread t = kernel().vp_manager.vp_to_thread( kernel().vp_manager.suggest_vp( gid ) );
+  Node* node = get_node_or_proxy( gid, t );
 
-  Node* node = local_nodes_[ t ].get_node_by_gid( gid );
+  return node;
+}
+
+Node* NodeManager::get_mpi_local_node_or_device_head( index gid )
```

Comment:
It's late and I might now be fully awake anymore, but why is this called `_head`? Can you please enlighten me? Thanks!